src/ProjectTemplates/BlazorWasm.ProjectTemplates/BlazorWasm-CSharp.Client.csproj.in

```diff
@@ -11,10 +11,25 @@
     <PackageReference Include="Microsoft.AspNetCore.Blazor.DevServer" Version="${MicrosoftAspNetCoreBlazorDevServerPackageVersion}" PrivateAssets="all" />
     <PackageReference Include="Microsoft.AspNetCore.Blazor.HttpClient" Version="${MicrosoftAspNetCoreBlazorHttpClientPackageVersion}" />
   </ItemGroup>
+
   <!--#if Hosted -->
   <ItemGroup>
     <ProjectReference Include="..\Shared\BlazorWasm-CSharp.Shared.csproj" />
   </ItemGroup>
+
   <!--#endif -->
+  <!--#if PWA -->
+  <Target Name="PublishServiceWorker" AfterTargets="CopyFilesToPublishDirectory">
+    <!-- Use the "published" version of the service worker in the output -->
+    <PropertyGroup>
+      <ServiceWorkerMoveFrom>$(PublishDir)$(BlazorPublishDistDir)service-worker.published.js</ServiceWorkerMoveFrom>
+      <ServiceWorkerMoveTo>$(PublishDir)$(BlazorPublishDistDir)service-worker.js</ServiceWorkerMoveTo>
+    </PropertyGroup>
+    <Move SourceFiles="$(ServiceWorkerMoveFrom)" DestinationFiles="$(ServiceWorkerMoveTo)" />
```

Comment:
Why not publish it without the `published.js` extension?

Reply:
We do, that's what this does. Or are you suggesting this should be implemented differently?

I've tried to keep the MSBuild logic here incredibly basic. For example, not assuming the developer even knows what an item group is. It's just properties, which should be familiar to any OO developer.

Reply:
I think that I agree with Pranav, I feel it is better to do this outside of the user project. I don't think that the logic here is simple. We introduce the user to several MSBuild concepts and properties and it is not straightforward to understand what is happening here.

We could do this in several different ways:
* Add a new package for doing service workers.
  * All the logic gets encapsulated there and we trigger it based on the presence of a service-worker.js file on your app.
  * The fact that you added the package is the gesture you want the SW.
  * Everything gets handled transparently for you.
* We put the logic inside blazor.build
  * Same deal, but no package.
  * We would need a property to turn it off
  * Would be turned on by default based on the presence of a given file.